metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/service/ProxiesService.java

```diff
@@ -394,6 +399,52 @@ public RecordsResponse getListOfFileContentsFromPluginExecution(MetisUserView me
     // Done.
     return new RecordsResponse(records);
   }
+  /**
+   * Get a record from the external resource based on o searchId, workflow execution and {@link PluginType}.
+   *
+   * @param metisUserView the user wishing to perform this operation
+   * @param workflowExecutionId the execution identifier of the workflow
+   * @param pluginType the {@link ExecutablePluginType} that is to be located inside the workflow
+   * @param searchId the searchId of the record we wish to obtain
+   * @return the record from the external resource
+   * @throws GenericMetisException can be one of:
+   * <ul>
+   * <li>{@link MCSException} if an error occurred while retrieving the records from the external
+   * resource</li>
+   * <li>{@link eu.europeana.metis.exception.UserUnauthorizedException} if the user is not
+   * authorized to perform this task</li>
+   * <li>{@link eu.europeana.metis.core.exceptions.NoWorkflowExecutionFoundException} if no workflow
+   * execution exists for the provided identifier</li>
+   * </ul>
+   */
+  public Record lookupIdFromUISClient(MetisUserView metisUserView, String workflowExecutionId , ExecutablePluginType pluginType, String searchId )
+      throws GenericMetisException {
+
+    // Get the right workflow execution and plugin type.
+    final Pair<WorkflowExecution, AbstractExecutablePlugin> executionAndPlugin = getExecutionAndPlugin(
+        metisUserView, workflowExecutionId, pluginType);
+    if (executionAndPlugin == null) {
+      throw new NoWorkflowExecutionFoundException(String
+          .format("No executable plugin of type %s found for workflowExecution with id: %s",
+              pluginType.name(), workflowExecutionId));
+    }
+    // Authorize
+//    authorizer.authorizeReadAllDatasets(metisUserView);
+
+    // lookup for eCloudId.
+    final CloudId cloudId;
+    try {
+      cloudId = uisClient.getCloudId(searchId,searchId);
+
+
+      if (cloudId != null) {
+        return Optional.ofNullable(getRecord(executionAndPlugin.getRight(), cloudId.getId())).orElse(null);
+      }
+    } catch (CloudException e) {
+      throw new GenericMetisException(String.format("Failed to lookup cloudId for searchId: %s", searchId), e);
```

Comment:
Should throw an instance of `ExternalTaskException` (which is a subclass of `GenericMetisException`).

Reply:
Done.